common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h

```diff
@@ -348,6 +348,106 @@ struct SLinkMeasurementsResponse11k {
     uint32_t dmg_link_adapt_ack_reference_timestamp;
 };
 
+enum eChannelScanResultMode : uint8_t {
```

Comment:
What @tomereli means, is that once enums are declared as `class` they are accessed using their type, so it can be removed from the elements.
For example, based on the change above, the usage will be `eChannelScanResultMode::NA`, `eChannelScanResultMode::AdHoc` and `eChannelScanResultMode::Infra`.